improver_tests/utilities/test_DurationSubdivision.py

```diff
@@ -0,0 +1,758 @@
+            {
+                "target_period": 3600,
+                "fidelity": 1800,
+                "night_mask": True,
+                "day_mask": False,
+            },
+            np.full((3, 3), 7200),  # Data in the input cube.
+            dt(2024, 6, 15, 21),  # Validity time
+            7200,  # Input period
+            np.array(
+                [
+                    [[3600, 1800, 0], [3600, 3600, 3600], [3600, 3600, 3600]],
+                    [[3600, 0, 0], [3600, 3600, 0], [3600, 3600, 3600]],
+                ],
+                dtype=np.float32,
+            ),  # Expected data in the output cube.
+            None,  # List of realization numbers if any
+            None,  # Expected exception
+        ),  # Night masking applies time dependent masking, meaning we get different data
+        # in each of the returned periods. The total in any period does not exceed the
+        # length of that period; this means that the total across periods is lower than
+        # in the input.
+        (
+            {
+                "target_period": 3600,
+                "fidelity": 1800,
+                "night_mask": False,
+                "day_mask": True,
+            },
+            np.full((3, 3), 7200),  # Data in the input cube.
+            dt(2024, 6, 15, 21),  # Validity time
+            7200,  # Input period
+            np.array(
+                [
+                    [[0, 1800, 3600], [0, 0, 0], [0, 0, 0]],
+                    [[0, 3600, 3600], [0, 0, 3600], [0, 0, 0]],
+                ],
+                dtype=np.float32,
+            ),  # Expected data in the output cube.
+            None,  # List of realization numbers if any
+            None,  # Expected exception
+        ),  # Day masking applies time dependent masking. This is the inverse of the test
+        # above, and we get a suitably inverted result.
+        (
+            {
+                "target_period": 3600,
+                "fidelity": 360,
+                "night_mask": True,
+                "day_mask": False,
+            },
+            np.full((3, 3), 7200),  # Data in the input cube.
+            dt(2024, 6, 15, 21),  # Validity time
+            7200,  # Input period
+            np.array(
+                [
+                    [[3600, 1440, 0], [3600, 3600, 3240], [3600, 3600, 3600]],
+                    [[2880, 0, 0], [3600, 3600, 0], [3600, 3600, 3600]],
+                ],
+                dtype=np.float32,
+            ),  # Expected data in the output cube.
+            None,  # List of realization numbers if any
+            None,  # Expected exception
+        ),  # Repeat the night masking test above but with higher fidelity, meaning that
+        # the subdivided data is split into shorter periods and the night mask
+        # applied more accurately. As a result we get fractions returned which are
+        # multiples of this fidelity period.
+        (
+            {
+                "target_period": 3600,
```

Comment:
Would it be necessary to include a test where the fidelity is higher than the target_period?

Reply:
There should be an exception for this, which I've now added, along with an associated test. Good spot.